### PR TITLE
ft(API): User can delete a booking

### PR DESCRIPTION
**Pull request by amkayondo**
#### What does this PR do?
Adds files that enable User can delete a booking
#### Description of Task to be completed?
Users can delete their booking.
#### How should this be manually tested?
send `DELETE` - `/bookings/<:bookingId>`
#### Any background context you want to provide?
`200` - OK
`400` - BAD REQUEST
`404` - NOT FOUND
#### What are the relevant pivotal tracker stories?
[#167912173](https://www.pivotaltracker.com/story/show/167912173)
#### Screenshots
Screenshot (68)
Screenshot (69)
Screenshot (65)
Screenshot (66)
Screenshot (67)

### CODE REVIEW

File: server/models/bookings.js [hunk 1]
```diff
@@ -30,12 +39,14 @@ export default class Book {
...
-    return parseInt(findTrip + seats);
+
+  async checkIfTripExists(bus_license_number, trip_date) {
+    const result = await newTrip.getTripBylience(bus_license_number, trip_date);
+    return result;
   }
 
-  async checkIfTripExists(trip_id) {
-    const result = await newTrip.getTripById(trip_id);
+  async checkIfTripExistsById(tripId) {
```
Comment (Kasulejoseph) [inline server/models/bookings.js]: I think `isTripIdExixts` would be a better method name here

File: server/models/bookings.js [hunk 2]
```diff
@@ -20,7 +20,16 @@ export default class Book {
 
   async decreaseNumberOfSeats(tripId, seatsAvailable, newSeats) {
     const newData = parseInt(seatsAvailable - newSeats);
+    const result = await db.updateBookings(tripId, newData);
+    return result;
+  }
+
+  async increaseNumberOfSeats(tripId, seatsAvailable, newSeats) {
```
Comment (Kasulejoseph) [inline server/models/bookings.js]: `decreaseNumberOfSeats` and `increaseNumberOfSeats` can be a single method try refactoring..

File: server/models/bookings.js [hunk 3]
```diff
@@ -20,7 +20,16 @@ export default class Book {
...
+    const newData = seatsAvailable + newSeats;
     console.log(newData);
+    console.log(seatsAvailable);
+    console.log(newSeats);
```
Comment (Kasulejoseph) [inline server/models/bookings.js]: Remove the debuger statements as well

File: server/models/bookings.js [hunk 4]
```diff
@@ -30,12 +39,14 @@ export default class Book {
     return allbookings;
   }
 
-  async increaseNumberOfSeats(findTrip, seats) {
-    return parseInt(findTrip + seats);
+
+  async checkIfTripExists(bus_license_number, trip_date) {
```
Comment (Kasulejoseph) [inline server/models/bookings.js]: `isTripExists` would be better

File: server/models/bookings.js [hunk 5]
```diff
@@ -47,12 +58,18 @@ export default class Book {
     return newTrip.findTripDate(Date);
   }
 
+  async deleteBookingByUser(booking_id, user_id){
+    const result = await db.deleteBooking(booking_id, user_id);
+    return result;
+  }
+
   async checkBookingUser(userid) {
     // return this.getAtllbookings.find(x => x.userId === userid);
```
Comment (Kasulejoseph) [inline server/models/bookings.js:57]: What's the purpose of this method? If it doesn't do anything please remove it

**Review by Kasulejoseph — CHANGES_REQUESTED** [review]

**Comment by Kasulejoseph** [comment]
In the response, you're having a typo `massage` instead of `message` please correct it

File: server/models/bookings.js (hunk 5 above)
Comment (amkayondo) [reply server/models/bookings.js:57]: I am going to use it when am getting user bookings

File: server/controllers/bookings/getAllBookings.js [hunk 6]
```diff
@@ -2,22 +2,23 @@ import jwt from 'jsonwebtoken';
...
+    if (allBookings.rowCount === 0) {
+      return resPonse.errorMessage(res, 404, 'No bookings made at the moment');
+    } resPonse.successData(res, 200, 'All bookings successfully fetched', allBookings.rows);
   }
+  const forUser = await book.getUserBookings(getUser.id);
+  if (forUser.length === 0) {
+    return resPonse.errorMessage(res, 200, 'You have made no bookings yet');
+  } resPonse.successData(res, 200, forUser);
+  // } catch (error) {
+  //   resPonse.errorMessage(res, 500, error.message);
```
Comment (Kasulejoseph) [inline server/controllers/bookings/getAllBookings.js:20]: Why are you commenting this out, remove it entirely if you think it doesn't add any value.

**Review by Kasulejoseph — CHANGES_REQUESTED** [review]

**Review by Kasulejoseph — APPROVED** [review]

**Pull request merged**